openquake/hazardlib/gsim/idriss_2014.py

```diff
@@ -0,0 +1,183 @@
+    for sites with Vs30 < 450 m/s
+    """
+    #: Supported tectonic region type is active shallow crust
+    DEFINED_FOR_TECTONIC_REGION_TYPE = const.TRT.ACTIVE_SHALLOW_CRUST
+
+    #: Supported intensity measure types are spectral acceleration,
+    #:and peak ground acceleration
+    DEFINED_FOR_INTENSITY_MEASURE_TYPES = set([
+        PGA,
+        SA
+    ])
+
+    #: Supported intensity measure component is orientation-independent
+    #: measure :attr:`~openquake.hazardlib.const.IMC.RotD50`
+    DEFINED_FOR_INTENSITY_MEASURE_COMPONENT = const.IMC.RotD50
+
+    #: Supported standard deviation types are total
+    DEFINED_FOR_STANDARD_DEVIATION_TYPES = set([
+        const.StdDev.TOTAL,
+    ])
+
+    #: Required site parameters is Vs30
+    REQUIRES_SITES_PARAMETERS = set(('vs30', ))
+
+    #: Required rupture parameters are magnitude, and rake.
+    REQUIRES_RUPTURE_PARAMETERS = set(('mag', 'rake'))
+
+    #: Required distance measure is Rrup
+    REQUIRES_DISTANCES = set(('rrup', ))
+
+    def get_mean_and_stddevs(self, sites, rup, dists, imt, stddev_types):
+        """
+        See :meth:`superclass method
+        <.base.GroundShakingIntensityModel.get_mean_and_stddevs>`
+        for spec of input and result values.
+        """
+        # extracting dictionary of coefficients specific to required
+        # intensity measure type.
+        C = self.COEFFS[imt]
+
+        mean = (self._get_magnitude_scaling_term(C, rup.mag) +
+                self._get_distance_scaling_term(C, rup.mag, dists.rrup) +
+                self._get_style_of_faulting_term(C, rup.rake) +
+                self._get_site_scaling_term(C, sites.vs30))
+
+        stddevs = self._get_stddevs(imt,
+                                    rup.mag,
+                                    len(dists.rrup),
+                                    stddev_types)
+        return mean, stddevs
+
+    def _get_magnitude_scaling_term(self, C, mag):
+        """
+        Returns the magnitude scaling term defined in equation 3
+        """
+        if mag < 6.75:
+            return C["a1_lo"] + C["a2_lo"] * mag + C["a3"] *\
+                ((8.5 - mag) ** 2.0)
+        else:
+            return C["a1_hi"] + C["a2_hi"] * mag + C["a3"] *\
+                ((8.5 - mag) ** 2.0)
+
+    def _get_distance_scaling_term(self, C, mag, rrup):
+        """
+        Returns the magnitude dependent distance scaling term
+        """
+        if mag < 6.75:
+            mag_factor = -(C["b1_lo"] + C["b2_lo"] * mag)
+        else:
+            mag_factor = -(C["b1_hi"] + C["b2_hi"] * mag)
+        return mag_factor * np.log(rrup + 10.0) + (C["gamma"] * rrup)
+
+    def _get_style_of_faulting_term(self, C, rake):
+        """
+        Only distinction is between reverse faulting events and
+        normal/strike-slip. Returns the style-of-faulting factor only for
+        reverse events
+        """
+        if (rake > 30.0) and (rake < 150.0):
+            return C["phi"]
```

Comment:
Yes they really do use the letter phi as the scaling coefficient for the style-of-faulting (see equation 3).